### PR TITLE
Update UI to re-sync with sharing changes

### PR DESCRIPTION
**Pull request by kevodwyer**

### CODE REVIEW

File: src/components/filesystem/index.js [hunk 1]
```diff
@@ -331,25 +332,43 @@ module.exports = {
                 that.currentDir = file.get();
                 that.updateFiles(refreshSharedWith);
             }).exceptionally(function(throwable) {
-                throwable.printStackTrace();
+                console.log(throwable.getMessage());
```
Comment (ianopolous) [inline src/components/filesystem/index.js:335]: any reason to not also do a stack trace?

Comment (kevodwyer) [reply src/components/filesystem/index.js:335]: stacktrace not available in js

Comment (ianopolous) [reply src/components/filesystem/index.js:335]: I've seen java stacktraces in the browser console?

File: src/components/filesystem/index.js [hunk 2]
```diff
@@ -331,25 +332,32 @@ module.exports = {
...
-            current.getChildren(that.getContext().crypto.hasher, that.getContext().network).thenApply(function(children){
-                var arr = children.toArray();
-                that.showSpinner = false;
-                that.files = arr.filter(function(f){
-                    return !f.getFileProperties().isHidden;
+            let that = this;
+            let context = this.getContext();
+            let directoryPath = peergos.client.PathUtils.directoryToPath(that.path);
+            context.getDirectorySharingState(directoryPath).thenApply(function(updatedSharedWithState) {
+                that.sharedWithState = updatedSharedWithState;
```
Comment (ianopolous) [inline src/components/filesystem/index.js:346]: I would wait to set this until the children are also retrieved

File: src/components/filesystem/index.js [hunk 3]
```diff
@@ -331,25 +332,32 @@ module.exports = {
...
+                current.getChildren(that.getContext().crypto.hasher, context.network).thenApply(function(children){
+                    var arr = children.toArray();
+                    that.showSpinner = false;
+                    that.files = arr.filter(function(f){
+                        return !f.getFileProperties().isHidden;
+                    });
+                    if (refreshSharedWith) {
```
Comment (ianopolous) [inline src/components/filesystem/index.js:353]: when do we not want to refreshSharedWith?

File: src/components/filesystem/index.js (hunk 1 above)
Comment (kevodwyer) [reply src/components/filesystem/index.js:335]: There is no printStackTrace() on the throwable object returned at least. Seems it isn't exposed. We don't call printStackTrace anywhere in the vanilla js code.

File: src/components/filesystem/index.js (hunk 3 above)
Comment (kevodwyer) [reply src/components/filesystem/index.js]: Simplified now

Comment (ianopolous) [reply src/components/filesystem/index.js]: Very nice

**Pull request merged**